### PR TITLE
Adding PIDName class

### PR DESCRIPTION
**Pull request by Sun-flow**
Plan to solve:

    Introduce class similar to BaseName, which takes inputs for species name, PID, and database info was acquired from
    PIDName(name=str,ID=str,database=str)
    Will place in base, and utilize in subject.species

### CODE REVIEW

File: examples/subject.json [hunk 1]
```diff
@@ -1,7 +1,11 @@
 {
    "describedBy": "https://raw.githubusercontent.com/AllenNeuralDynamics/aind-data-schema/main/src/aind_data_schema/subject.py",
-   "schema_version": "0.3.0",
-   "species": "Mus musculus",
+   "schema_version": "0.3.1",
+   "species": {
+      "name": "Mus musculus",
+      "ID": "10090",
+      "database": "NCBI"
```
Comment (saskiad) [inline examples/subject.json:7]: Let's have the database be the url for the taxonomy so there's absolutely no confusion

File: src/aind_data_schema/base.py [hunk 2]
```diff
@@ -51,6 +51,12 @@ class BaseName(AindModel):
     name: str
     abbreviation: str
 
+class PIDName(AindModel):
+    """Model tracking a species name, PID, and database PID was acquired from"""
+    name: str
+    ID: str
```
Comment (saskiad) [inline src/aind_data_schema/base.py:57]: it looks like ID might be a specific thing in pydantic (hence the different color). Maybe call this "identifier"?

File: src/aind_data_schema/base.py [hunk 3]
```diff
@@ -51,6 +51,12 @@ class BaseName(AindModel):
     name: str
     abbreviation: str
 
+class PIDName(AindModel):
+    """Model tracking a species name, PID, and database PID was acquired from"""
+    name: str
```
Comment (saskiad) [inline src/aind_data_schema/base.py:56]: ah... some syntax here:

name: str = Field(..., title="Name") 

yeah, David didn't do that for the BaseName... but generally we want to stick with this syntax. For database, add a description that instructs users something like "URL for the external database"

File: src/aind_data_schema/base.py [hunk 4]
```diff
@@ -51,6 +51,12 @@ class BaseName(AindModel):
...
+    ID: str
+    database: str
```
Comment (saskiad) [inline src/aind_data_schema/base.py:58]: thought I had from below - I think maybe add database_name and database_url

**Review by saskiad — CHANGES_REQUESTED** [review]
see comments

**Review by dyf — APPROVED** [review]
love it

File: src/aind_data_schema/base.py [hunk 5]
```diff
@@ -47,10 +49,15 @@ def __call__(cls, value, *args, **kw):
 
 class BaseName(AindModel):
     """A simple model associating a name with an abbreviation"""
-
     name: str
     abbreviation: str
 
+class PIDName(AindModel):
+    """Model tracking a species name, PID, and database PID was acquired from"""
```
Comment (saskiad) [inline src/aind_data_schema/base.py:56]: "Model for associate a name with a persistent identifier (PID) and the registry for that PID"

Also, add a line after the description (just to match our formatting) :)

File: src/aind_data_schema/base.py [hunk 6]
```diff
@@ -47,10 +49,15 @@ def __call__(cls, value, *args, **kw):
...
+    name: str = Field(..., title="Name")
+    registry: str = Field(..., title="Registry")
```
Comment (saskiad) [inline src/aind_data_schema/base.py:58]: I think for now this is good - I think soon we might make the registry an ENUM of the databases we use. But we can do that when we apply PIDName to other things!

**Review by saskiad — CHANGES_REQUESTED** [review]
one minor edit. Otherwise it looks good!

File: src/aind_data_schema/base.py (hunk 4 above)
Comment (saskiad) [reply src/aind_data_schema/base.py]: Noting that we've decided not to add the registry url

**Review by saskiad — APPROVED** [review]
Awesome!!

**Comment by Sun-flow** [comment]
close #294

**Pull request merged**